src/pages/Accounts/SendConfirmation_updated.tsx

```diff
@@ -0,0 +1,955 @@
+  heightPercentageToDP as hp,
+} from 'react-native-responsive-screen';
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  transferST1,
+  transferST2,
+  clearTransfer,
+  fetchBalanceTx,
+  alternateTransferST2,
+} from '../../store/actions/accounts';
+import { UsNumberFormat } from '../../common/utilities';
+import BottomSheet from 'reanimated-bottom-sheet';
+import Slider from 'react-native-slider';
+import BottomInfoBox from '../../components/BottomInfoBox';
+import ModalHeader from '../../components/ModalHeader';
+import SendConfirmationContent from './SendConfirmationContent';
+import RecipientComponent from './RecipientComponent';
+import { createRandomString } from '../../common/CommonFunctions/timeFormatter';
+import moment from 'moment';
+import {
+  REGULAR_ACCOUNT,
+  TEST_ACCOUNT,
+  SECURE_ACCOUNT,
+} from '../../common/constants/serviceTypes';
+import RelayServices from '../../bitcoin/services/RelayService';
+import {
+  INotification,
+  notificationTag,
+  notificationType,
+} from '../../bitcoin/utilities/Interface';
+import { withNavigationFocus } from 'react-navigation';
+import { connect } from 'react-redux';
+import idx from 'idx';
+
+interface SendConfirmationStateTypes {
+  switchOn: boolean;
+  CurrencyCode: string;
+  totalAmount: any;
+  sliderValue: any;
+  sliderValueText: string;
+  exchangeRates: any;
+  SelectedContactId: any;
+  transfer: {},
+  loading: {}
+}
+
+interface SendConfirmationPropsTypes {
+  navigation: any;
+  accounts: any[];
+  WALLET_SETUP: any;
+  trustedContactsService: any;
+  exchangeRates: any;
+  fetchBalanceTx: any;
+  clearTransfer: any;
+  alternateTransferST2: any;
+  transferST2: any;
+}
+class SendConfirmation_updated extends Component<SendConfirmationPropsTypes, SendConfirmationStateTypes> {
+  focusListener: any;
+  appStateListener: any;
+  firebaseNotificationListener: any;
+  notificationOpenedListener: any;
+  NoInternetBottomSheet: any;
+  unsubscribe: any;
+  serviceType: any;
+  sweepSecure: any;
+  spendableBalance: any;
+  transfer: any;
+  loading: any;
+  viewRef: any;
+
+  constructor(props) {
+    super(props);
+    this.serviceType = this.props.navigation.getParam('serviceType')
+    this.sweepSecure = props.navigation.getParam('sweepSecure');
+    this.spendableBalance = props.navigation.getParam('spendableBalance');
+    this.viewRef = React.createRef();
+    this.state = {
+      switchOn: true,
+      CurrencyCode: 'USD',
+      totalAmount: 0,
+      sliderValue: 0,
+      sliderValueText: 'Low Fee',
+      exchangeRates: this.props.exchangeRates,
+      SelectedContactId: 0,
+      transfer: {},
+      loading: {}
+    }
+  }
+
+  componentDidMount = () =>{
+    this.onChangeInTransfer();
+    let { accounts } = this.props;
+    if (accounts[this.serviceType].transfer.details) {
+      let totalAmount = 0;
+      accounts[this.serviceType].transfer.details.map((item) => {
```

Comment:
Please use idx here since nested objects are being accessed, also check if array is defined before calling map